### PR TITLE
Add source and behaviour information to docs chunk metadata

### PR DESCRIPTION
**Pull request by jonatanklosko**
Implements suggestions from https://github.com/erlang/otp/pull/8945.

### CODE REVIEW

File: lib/elixir/src/elixir_erl.erl [hunk 1]
```diff
@@ -39,12 +39,14 @@ debug_info(_, _, _, _) ->
...
-get_ann([{generated, true} | T], _, Line) -> get_ann(T, true, Line);
-get_ann([{line, Line} | T], Gen, _) when is_integer(Line) -> get_ann(T, Gen, Line);
-get_ann([_ | T], Gen, Line) -> get_ann(T, Gen, Line);
-get_ann([], Gen, Line) -> erl_anno:set_generated(Gen, erl_anno:new(Line)).
+get_ann([{generated, true} | T], _, Line, Column) -> get_ann(T, true, Line, Column);
+get_ann([{line, Line} | T], Gen, _, Column) when is_integer(Line) -> get_ann(T, Gen, Line, Column);
+get_ann([{column, Column} | T], Gen, Line, _) when is_integer(Column) -> get_ann(T, Gen, Line, Column);
+get_ann([_ | T], Gen, Line, Column) -> get_ann(T, Gen, Line, Column);
+get_ann([], Gen, Line, undefined) -> erl_anno:set_generated(Gen, erl_anno:new(Line));
+get_ann([], Gen, Line, Column) -> erl_anno:set_generated(Gen, erl_anno:new({Line, Column})).
```
Comment (jonatanklosko) [inline lib/elixir/src/elixir_erl.erl:49]: Previously this ignored column from meta, that's why the assertions in warning_test.exs now include a column as well :)

File: lib/elixir/src/elixir_erl.erl [hunk 2]
```diff
@@ -243,13 +247,29 @@ is_macro(defmacro)  -> true;
...
+  % we take the one with earliest line.
+
+  LineComparator = fun
+    ({Def1, Meta1}, {Def1, Meta2}) -> ?line(Meta1) =< ?line(Meta2);
+    ({Def1, _Meta1}, {Def2, _Meta2}) -> Def1 =< Def2
+  end,
+
+  UniqFun = fun({Def, _Meta}) -> Def end,
+
+  lists:uniq(UniqFun, lists:sort(LineComparator, Entries)).
```
Comment (josevalim) [inline lib/elixir/src/elixir_erl.erl:261]: I don't think we actually need uniqueness here. We only need sorting but `lists:usort` is faster than `lists:sort`, which is why we called `usort` before.  I don't think we actually expect duplicate tuples in there!

**Comment by josevalim** [comment]
Just some small nitpicks. Once the Erlang/OTP PRs have been merged, I will merge this one. :)

File: lib/elixir/src/elixir_erl.erl (hunk 2 above)
Comment (jonatanklosko) [reply lib/elixir/src/elixir_erl.erl:261]: There can be duplicates for callbacks with multiple clauses. Looks like that's not the case for definitions though!

Comment (jonatanklosko) [reply lib/elixir/src/elixir_erl.erl]: Updated!

**Comment by josevalim** [comment]
I will merge this, since the Erlang/OTP PR for the Docs reference has been merged, although we are waiting on updates on how we should deal with ranges, based on https://github.com/erlang/otp/pull/8966 and https://github.com/erlang/otp/pull/8975.

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**